src/IpAddress.php

```diff
@@ -62,19 +37,11 @@ public function __construct(
         $attributeName = null,
         array $headersToInspect = []
     ) {
-        if ($checkProxyHeaders && $trustedProxies === null) {
-            throw new \InvalidArgumentException('Use of the forward headers requires an array for trusted proxies.');
-        }
-
-        $this->checkProxyHeaders = $checkProxyHeaders;
-        $this->trustedProxies = $trustedProxies;
-
         if ($attributeName) {
             $this->attributeName = $attributeName;
         }
-        if (!empty($headersToInspect)) {
-            $this->headersToInspect = $headersToInspect;
-        }
+
+        $this->parser = new IpAddressParser($checkProxyHeaders, $trustedProxies, $headersToInspect);
```

Comment:
I could see the potential desire to have this as an optional parameter on the constructor, but I figured I'd leave that up to you guys.

Reply:
I'm not too worried about that as if someone wants to use their own parser, they can write their own middleware easily enough as it doesn't do much now!